### PR TITLE
Make Celery worker the root process

### PR DESCRIPTION
**Pull request by hacktobeer**
### Description of the change
This PR strips away layers of complexity and makes the Celery worker the root process in the container. This will benefit the reliability and stability of the system.
* will eliminate the race condition we had in k8s
* will eliminate the need for scaledown locks in both Turbinia code and helm charts (follow-up PR to remove that code)
* Will in general make debugging easier as there are less Turbinia wrapper classes involved.

This PR also:
* adds hot reloading of the Celery worker through watchdog without the need for Python binary injection hacks for development purposes (see Dockerfile)
* set default log level for the worker to INFO which can be changed to DEBUG by the setting the TURBINIA_DEBUG=1 environment variable.

### Applicable issues

-fixes #

### Checklist

- [x] All tests were successful.
- [ ] Unit tests added.
- [ ] Documentation updated.

### CODE REVIEW

**Comment by hacktobeer** [comment]
@jleaniz @wajihyassine Besides the normal code review I would like you to review this PR with a bigger scope in your mind as well if you have time.

E.g. Do we need to cleanup move code or documents? Can we tune the Celery settings? Can we introduce different task types (as we only hard code one task (task-runner) for all jobs :(.

**Comment by hacktobeer** [comment]
@wajihyassine Do we use any of the arguments from start.sh in the helm charts? As with this PR we are *not* using start.sh and are not passing in those args.
https://github.com/google/turbinia/blob/master/docker/worker/start.sh

**Review by jleaniz — APPROVED** [review]
LGTM. Thanks for these changes. We'll remove the run_wrapper lock in a separate PR.

File: turbinia/turbiniactl.py [hunk 1]
```diff
@@ -182,12 +182,9 @@ def process_args(args):
...
   elif args.command == 'celeryworker':
-    initialize_debugmode_if_requested()
-    # pylint: disable=import-outside-toplevel
-    from turbinia.worker import TurbiniaCeleryWorker
-    worker = TurbiniaCeleryWorker(
-        jobs_denylist=args.jobs_denylist, jobs_allowlist=args.jobs_allowlist)
-    worker.start()
+    log.error(
+        '''Deprecated: Please start celery worker
+           directly with `celery -A turbinia.app worker''')
```
Comment (wajihyassine) [inline turbinia/turbiniactl.py:187]: Optional small nit add an extra ` at the end of worker

**Review by wajihyassine — APPROVED** [review]
LGTM, thank you for these changes! It would be interesting seeing how we can use this to split up tasks differently especially the more resource intensive tasks vs not

**Pull request merged**